executeSQL(): crashes when executing no-args query

Here is the setup:

	db.executeSQL()
	        .withQuery(RawQuery.builder()
	                .query("DROP TABLE IF EXISTS items")
	                .build())
	        .prepare()
	        .executeAsBlocking();

This fails with following exception:

     Caused by: java.lang.IllegalArgumentException: Empty bindArgs
            at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1651)
            at com.pushtorefresh.storio.sqlite.impl.DefaultStorIOSQLite$InternalImpl.executeSQL(DefaultStorIOSQLite.java:286)
            at com.pushtorefresh.storio.sqlite.operations.execute.PreparedExecuteSQL.executeAsBlocking(PreparedExecuteSQL.java:53)

This happens, because overloaded `android.database.sqlite.SQLiteDatabase#execSQL(java.lang.String, java.lang.Object[])` does not accept null `bindArgs` (as per android-23 source), while `DefaultStorIOSQLite` implementation passes `nullableArrayOfStrings(rawQuery.args())` as bindArgs. 

I also tried a workaround:

	db.executeSQL()
	        .withQuery(RawQuery.builder()
	                .query("DROP TABLE IF EXISTS ?")
	                .args("items")
	                .build())
	        .prepare()
	        .executeAsBlocking();

This fails with the following exception:

     Caused by: android.database.sqlite.SQLiteException: near "?": syntax error (code 1): , while compiling: DROP TABLE IF EXISTS ?
            at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
            at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:889)
            at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:500)
            at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
            at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
            at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
            at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1672)
            at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1653)
            at com.pushtorefresh.storio.sqlite.impl.DefaultStorIOSQLite$InternalImpl.executeSQL(DefaultStorIOSQLite.java:286)
            at com.pushtorefresh.storio.sqlite.operations.execute.PreparedExecuteSQL.executeAsBlocking(PreparedExecuteSQL.java:53)

I guess, binding is supported for values only (not for table or column names). So, I also doubt code in [this README section](https://github.com/pushtorefresh/storio/blob/master/docs/StorIOSQLite.md#4-execsql-operation) works. Guys, please, take a look.

--- a/storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/ui/fragment/TweetsFragment.java
+++ b/storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/ui/fragment/TweetsFragment.java
@@ -147,7 +147,7 @@ public class TweetsFragment extends BaseFragment {
         tweets.add(Tweet.newTweet("Google", "Android be together not the same: AOSP, AOSP + Google Apps, Samsung Android"));
         tweets.add(Tweet.newTweet("Reddit", "Now we can send funny gifs directly into your brain via Oculus Rift app!"));
         tweets.add(Tweet.newTweet("ElonMusk", "Tesla Model S OTA update with Android Auto 5.2, fixes for memory leaks"));
-        tweets.add(Tweet.newTweet("AndroidWeekly", "Special issue #1: StorIO — forget about SQLiteDatabase, ContentResolver APIs, ORMs sucks!"));
+        tweets.add(Tweet.newTweet("AndroidWeekly", "Special issue #1: StorIO — forget about SQLiteDatabase, ContentResolver APIs, ORMs suck!"));
         tweets.add(Tweet.newTweet("Apple", "Yosemite update: fixes for Wifi issues, yosemite-wifi-patch#142"));
 
         // Looks/reads nice, isn't it?

--- a/storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLite.java
+++ b/storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLite.java
@@ -281,12 +281,18 @@ public class DefaultStorIOSQLite extends StorIOSQLite {
         @WorkerThread
         @Override
         public void executeSQL(@NonNull RawQuery rawQuery) {
-            sqLiteOpenHelper
-                    .getWritableDatabase()
-                    .execSQL(
-                            rawQuery.query(),
-                            nullableArrayOfStrings(rawQuery.args())
-                    );
+            if (rawQuery.args().isEmpty()) {
+                sqLiteOpenHelper
+                        .getWritableDatabase()
+                        .execSQL(rawQuery.query());
+            } else {
+                sqLiteOpenHelper
+                        .getWritableDatabase()
+                        .execSQL(
+                                rawQuery.query(),
+                                rawQuery.args().toArray(new String[rawQuery.args().size()])
+                        );
+            }
         }
 
         /**

--- a/storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLiteTest.java
+++ b/storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/impl/DefaultStorIOSQLiteTest.java
@@ -252,13 +252,11 @@ public class DefaultStorIOSQLiteTest {
     }
 
     @Test
-    public void shouldPassSQLToExecSQL() {
+    public void shouldPassSQLWithArgsToExecSQL() {
         SQLiteOpenHelper sqLiteOpenHelper = mock(SQLiteOpenHelper.class);
-
         SQLiteDatabase sqLiteDatabase = mock(SQLiteDatabase.class);
 
-        when(sqLiteOpenHelper.getWritableDatabase())
-                .thenReturn(sqLiteDatabase);
+        when(sqLiteOpenHelper.getWritableDatabase()).thenReturn(sqLiteDatabase);
 
         StorIOSQLite storIOSQLite = DefaultStorIOSQLite.builder()
                 .sqliteOpenHelper(sqLiteOpenHelper)
@@ -275,6 +273,30 @@ public class DefaultStorIOSQLiteTest {
 
         verify(sqLiteOpenHelper).getWritableDatabase();
         verify(sqLiteDatabase).execSQL(eq(rawQuery.query()), eq(new String[]{"arg1", "arg2"}));
+        verifyNoMoreInteractions(sqLiteOpenHelper, sqLiteDatabase);
+    }
+
+    @Test
+    public void shouldPassSQLWithoutArgsToExecSQL() {
+        SQLiteOpenHelper sqLiteOpenHelper = mock(SQLiteOpenHelper.class);
+        SQLiteDatabase sqLiteDatabase = mock(SQLiteDatabase.class);
+
+        when(sqLiteOpenHelper.getWritableDatabase()).thenReturn(sqLiteDatabase);
+
+        StorIOSQLite storIOSQLite = DefaultStorIOSQLite.builder()
+                .sqliteOpenHelper(sqLiteOpenHelper)
+                .build();
+
+        RawQuery rawQuery = RawQuery.builder()
+                .query("DROP TABLE IF EXISTS someTable")
+                .build(); // No args!
+
+        storIOSQLite
+                .internal()
+                .executeSQL(rawQuery);
+
+        verify(sqLiteOpenHelper).getWritableDatabase();
+        verify(sqLiteDatabase).execSQL(eq(rawQuery.query()));
         verifyNoMoreInteractions(sqLiteOpenHelper, sqLiteDatabase);
     }
 

--- a/storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/integration/ExecSQLTest.java
+++ b/storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/integration/ExecSQLTest.java
@@ -1,0 +1,26 @@
+package com.pushtorefresh.storio.sqlite.integration;
+
+import com.pushtorefresh.storio.sqlite.BuildConfig;
+import com.pushtorefresh.storio.sqlite.queries.RawQuery;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+public class ExecSQLTest extends BaseTest {
+
+    @Test
+    public void execSQLWithEmptyArgs() {
+        // Should not throw exceptions!
+        storIOSQLite
+                .executeSQL()
+                .withQuery(RawQuery.builder()
+                        .query("DROP TABLE IF EXISTS no_such_table") // we don't want to really delete table
+                        .build())
+                .prepare()
+                .executeAsBlocking();
+    }
+}

